pkg/prometheus/server/operator.go

```diff
@@ -793,28 +793,33 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return fmt.Errorf("synchronizing web config secret failed: %w", err)
 	}
 
-	if err := c.createOrUpdateThanosConfigSecret(ctx, p); err != nil {
-		return fmt.Errorf("failed to reconcile Thanos config secret: %w", err)
-	}
+	svcClient := c.kclient.CoreV1().Services(p.Namespace)
+	selectorLabels := makeSelectorLabels(p.Name)
 
-	// Reconcile the governing service.
-	svc := prompkg.BuildStatefulSetService(
-		governingServiceName,
-		map[string]string{"app.kubernetes.io/name": "prometheus"},
-		p,
-		c.config,
-	)
+	if p.Spec.ServiceName != nil {
+		if err := prompkg.EnsureCustomGoverningService(ctx, p, svcClient, selectorLabels); err != nil {
+			return fmt.Errorf("validating custom service failed: %w", err)
```

Comment:
(nit) We can probably avoid wrapping the returned error which has enough information already IMHO.
```suggestion
			return err
```